src/ert/_c_wrappers/enkf/subst_config.py

```diff
@@ -1,114 +1,84 @@
 import os.path
-from typing import Optional
+from datetime import date
+from typing import Dict
 
-from cwrap import BaseCClass
-
-from ert._c_wrappers import ResPrototype
+from ert._c_wrappers.config import ConfigContent
 from ert._c_wrappers.enkf.config_keys import ConfigKeys
 from ert._c_wrappers.util import SubstitutionList
 
 
-class SubstConfig(BaseCClass):
-    TYPE_NAME = "subst_config"
-    _alloc = ResPrototype("void* subst_config_alloc(config_content,int)", bind=False)
-    _alloc_full = ResPrototype("void* subst_config_alloc_full(subst_list)", bind=False)
-    _free = ResPrototype("void  subst_config_free(subst_config)")
-    _get_subst_list = ResPrototype(
-        "subst_list_ref subst_config_get_subst_list( subst_config )"
-    )
-
+class SubstConfig:
     def __init__(
-        self, config_content=None, config_dict=None, num_cpu: Optional[int] = None
+        self,
+        defines: Dict[str, str],
+        data_kw: Dict[str, str],
+        runpath_file_name: str,
+        num_cpu: int,
     ):
-        if not (config_content is not None) ^ (config_dict is not None):
-            raise ValueError(
-                "SubstConfig must be instansiated with exactly one"
-                " of config_content or config_dict"
-            )
+        subst_list = SubstitutionList()
```

Comment:
Since `SubstConfig` really only contains a `SubstitutionList` . Can't we just have `ResConfig.substitution_list` ? All inserting of `CONFIG_PATH`, `RUNPATH_FILE`, `DATE`, etc. can be moved to `ResConfig.substitution_list_from_dict`.

Reply:
`substitution_list_from_dict` becomes `SubstConfig.from_dict` combined with `Substconfig.__init__`.

Reply:
I see! yeah working on the next PR right now, it's gonna have these changes, where stuff is moved to res config and it's a "simple" substitution list, subst config disappears.